client/src/components/v3/status/CameraStatus.tsx

```diff
@@ -0,0 +1,61 @@
+import React from 'react';
+import { Col, Table } from 'react-bootstrap';
+import OnlineStatusPill from 'components/common/OnlineStatusPill';
+
+export interface CameraStatusProps {
+  cameraName: string;
+  online: boolean;
+  ip: string | null;
+  battery: number | null;
+  videoFeedStatus: boolean | null;
+}
```

Comment:
Here's a slightly neater thing you can do taking advantage of TypeScripts types. (untested)

```typescript
interface OnlineCameraState {
  cameraName: string;
  online: true;
  ip: string;
  battery: number;
  videoFeedStatus: boolean;
}
interface OfflineCameraState {
  cameraName: string;
  online: false;
}
type CameraStatus = OnlineCameraState | OfflineCameraState;
export interface CameraStatusProps extends CameraStatus {};
```

By doing this, typescript will realise that when `online` is `true`, the remaining fields must be present, and so you don't have to handle the cases when they're empty.

Reply:
We will have to handle the `null` case anyway, as `online ===  true` does not guarantee that `ip, battery, videoFeedStatus` will be non-null. The `null` indicates that the API may not have a response (e.g. not connected) rather than the values are not present.

Also, in trying to implement this, I think it makes it a lot more complicated, especially in the container.

Reply:
`ip`, `battery` and `videoFeedStatus` can only be null if you add `| null` to their type, right? Otherwise I have a big misunderstanding with typescript

Reply:
Yes, that's correct. However, those values are, presumably, from the API so they will have `| null` as the return type that will have to be handled either before or in the `CameraStatus` component.
Using the `OnlineCameraState` would imply that the `null` values have been handled before passing to the component (e.g. with default values), but it seems like it would make more sense for the component to handle them.